implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpHttpMetricsSender.java

```diff
@@ -41,19 +35,23 @@ public OtlpHttpMetricsSender(HttpSender httpSender) {
 
     /**
      * Send a batch of OTLP Protobuf format metrics to an OTLP HTTP receiver.
-     * @param address address of the OTLP HTTP receiver to which metrics will be sent
-     * @param metricsData OTLP protobuf encoded batch of metrics
-     * @param headers metadata to send as headers with the metrics data
-     * @throws Throwable when there is an exception in sending the metrics; the caller
+     * @param request metrics request to publish
+     * @throws Exception when there is an exception in sending the metrics; the caller
      * should handle this in some way such as logging the exception
      */
     @Override
-    public void send(String address, byte[] metricsData, Map<String, String> headers) throws Throwable {
-        HttpSender.Request.Builder httpRequest = this.httpSender.post(address)
+    public void send(Request request) throws Exception {
+        HttpSender.Request.Builder httpRequest = this.httpSender.post(request.getAddress())
             .withHeader("User-Agent", userAgentHeader)
-            .withContent("application/x-protobuf", metricsData);
-        headers.forEach(httpRequest::withHeader);
-        HttpSender.Response response = httpRequest.send();
+            .withContent("application/x-protobuf", request.getMetricsData());
+        request.getHeaders().forEach(httpRequest::withHeader);
+        HttpSender.Response response;
+        try {
+            response = httpRequest.send();
+        }
+        catch (Throwable e) {
+            throw new Exception(e);
```

Comment:
Since the `HttpSender` throws Throwable, we need to catch it here. Wrapping in `Exception` feels weird but I'm not sure what's better to do.